osu.Game/Extensions/ContainerExtensions.cs

```diff
@@ -0,0 +1,215 @@
+﻿// Copyright (c) ppy Pty Ltd <contact@ppy.sh>. Licensed under the MIT Licence.
+// See the LICENCE file in the repository root for full licence text.
+
+using System;
+using osu.Framework.Extensions.EnumExtensions;
+using osu.Framework.Graphics;
+using osu.Framework.Graphics.Containers;
+using osuTK;
+
+namespace osu.Game.Extensions
+{
+    public static class ContainerExtensions
+    {
+        /// <summary>
+        /// Prevents this container's children from transforming when the container transforms.
+        /// </summary>
+        /// <param name="target">The target to prevent its children from transforming.</param>
+        /// <param name="parent">The parent of the target.</param>
+        /// <param name="previousParentRotation">The parent's rotation in the previous update cycle.</param>
+        /// <param name="previousParentScale">The parent's scale in the previous update cycle.</param>
+        public static void KeepChildrenUpright(this Container target, Container parent, float previousParentRotation, Vector2 previousParentScale)
+        {
+            float parentRotation = parent.Rotation;
+            Vector2 parentScale = parent.Scale;
+
+            foreach (Drawable child in target)
+            {
+                child.RotateTo(-parentRotation);
+
+                int parentRotationInQuarterTurns = (int)Math.Floor(parentRotation / 90);
+                if (parentRotationInQuarterTurns % 2 != 0)
+                    child.ScaleTo(new Vector2(parentScale.Y, parentScale.X));
+                else
+                    child.ScaleTo(parentScale);
+
+                int previousParentRotationInQuarterTurns = (int)Math.Floor(previousParentRotation / 90);
+                int rotationsNeeded = parentRotationInQuarterTurns - previousParentRotationInQuarterTurns;
+                if (rotationsNeeded < 0)
+                    rotationsNeeded += 4;
```

Comment:
This implicitly assumes that `rotationsNeeded` is never below -4 to work correctly, which I don't know if it's right. I'm pretty sure `Rotation` can be any value as long as it fits into a `float`, it doesn't necessarily have to be in [-180, 180] or what have you.

The universal fix here would be to do something like

```csharp
if (rotationsNeeded < 0)
    rotationsNeeded = 4 + rotationsNeeded % 4;
```